app/dependencies/utils.py

```diff
@@ -1,18 +1,122 @@
 import json
 import logging
 import os
+import time
 from datetime import datetime
 
-import openai
 import tiktoken
 from dateutil.relativedelta import relativedelta
 from requests import Session
 from requests.exceptions import ConnectionError, Timeout, TooManyRedirects
 
-from app.dependencies.redis_client import set_redis_data
+from app.dependencies.redis_client import (
+    get_hexdata_from_redis,
+    set_redis_data,
+)
 
 
-def load_hex():
+def num_tokens_from_string(string: str, encoding_name: str) -> int:
+    """Returns the number of tokens in a text string."""
+    encoding = tiktoken.get_encoding(encoding_name)
+    num_tokens = len(encoding.encode(string))
+    return num_tokens
+
+
+def store_currency_in_redis():
+    url = "https://pro-api.coinmarketcap.com/v1/cryptocurrency/listings/latest"
+
+    rng = [1, 5001, 10001]
+    count = 1
+    for i in rng:
+        parameters = {"start": i, "limit": 5000, "convert": "USD", "aux": "cmc_rank"}
+
+        headers = {
+            "Accepts": "application/json",
+            "X-CMC_PRO_API_KEY": os.getenv("COIN_MARKET_CAP_KEY"),
+        }
+
+        session = Session()
+        session.headers.update(headers)
+
+        try:
+            response = session.get(url, params=parameters)
+            json_data = json.loads(response.text)
+
+            if count == 1:
+                set_redis_data("first_hexdata", json_data)
```

Comment:
instead of if count == i

why not something like

set_redis_data(f"{i}_data", json_data)

Reply:
this does not look good. 
if count == 1:
set_redis_data("first_hexdata", json_data)
elif count == 2:
set_redis_data("second_hexdata", json_data)
else:
set_redis_data("third_hexdata", json_data)

yes but simply do it like this

set_redis_data(f"{i}_hexdata", json_data)

Reply:
The i had three numbers like 1,5001,10001 so I avoided it but I'll correct it then it will be like. "1__hexdata", "5001_hexdata"

Reply:
is it all hexdata?? all variable function, key names should be appropriate
@Abdullah171